### PR TITLE
feat(dashboard-v2): unified home-screen activity log + editable categories, icon, mobile tabs

### PR DESCRIPTION
**Pull request by nadvolod**
## Summary

Unifies the dashboard-v2 home screen so **all** activity lives in one chronological feed, plus a batch of targeted fixes the user requested.

**Behavior changes (user-visible):**
- **Unified activity feed** — Morning Log and Reflections now appear as summary cards in the same activity feed as money/focus entries (desktop right column + mobile recent list). Morning Log shows `Sleep {score} · {h}h{m}m · N supplements · K habits`; Reflection shows `N of M answered`. The Three-to-Thrive panel still shows inline as before.
- **Tap any entry → read-only detail sheet** — clicking/tapping (or Enter/Space on) any feed row opens a detail sheet with the full record. Morning/Reflection details have an **Edit** button that opens their existing drawers; money/focus details are read-only.
- **Editable / removable categories** — Supplements, Habits, and Sleep-Environment custom fields can now be renamed and removed in the Morning Log drawer (previously add-only). Changes affect the template/future logs only; past notes are untouched.
- **Icon** — the Morning Log opener changed from `Moon` (read like a dark-mode toggle) to `NotebookPen`.
- **Removed `+ Call` / `+ Demo`** quick actions everywhere (mobile Quick Log, desktop pipeline card presets, ⌘K palette + its hint). Kept `+ Train` and `+ FU`.
- **Mobile Insights & Review tabs** now render the real `InsightsTab` / `ReviewTab` content instead of a dead-end placeholder.

**Backend / contracts:**
- `/api/health-notes` `update-templates` gains `editHabit` and `editEnvironmentField` operations (supplement edit + all add/remove already existed). Validation mirrors `editSupplement`; uniform success logging across template ops.

## User Flow Coverage

**Happy paths**
- Log a morning entry → summary card appears in feed → tap → full sleep detail; tap Edit → drawer reopens.
- Save a reflection → "Reflection" card appears in feed.
- Add/rename/remove a supplement, habit, or env field in the drawer → persists across reload.
- Mobile: tap Insights / Review in bottom nav → real tab content renders.
- Mobile: tap `+ Train` → trained metric increments.

**Failure / edge paths**
- Missing record on tap (e.g. stale refKey) → detail sheet shows a graceful "No data for this entry" fallback (no crash).
- Source-less / optimistic rows → tap is a no-op (sheet doesn't open with empty data).
- Partial morning log (null sleep metrics) → summary omits the score and duration chips; detail shows `—` per field (NaN-safe).
- Rename to an existing name / empty name → API returns 400 with a clear message; case-insensitive duplicate guard.
- Removing a template item does **not** rewrite past notes (history preserved).
- Cross-day ordering uses full epoch `tsMs` so morning/reflection cards sort correctly across day boundaries.
- E2E rows (test-seeded) remain filtered out of the feed.

## Evidence (CLI)

Local verification on the branch:
- `npx tsc --noEmit` → clean
- `npm run lint` → **0 errors** (82 pre-existing warnings)
- `npm test` → **923 passing** across 46 suites. The only 2 failing suites are pre-existing **env-gated integration tests** (`storage.integration` needs a DB; `monthly-review-api.integration` needs `SYNC_API_KEY`) that fail loudly by design when secrets are absent and pass in CI.
- `npm run build` → succeeds (`/dashboard` prerenders)
- New/updated unit tests: `derive.test.ts` (morning/reflection mappers + merge), `EditableItemList.test.tsx`, `ActivityDetailSheet.test.tsx`, `ActivityFeed.test.tsx` (click + keyboard), `MobileLayout.test.tsx` (real mobile tabs), `route.test.ts` + `route.editremove.test.ts` (rename ops, tracker + route level).
- E2E: 5 Playwright flows added (`morning-log.spec.ts`, `dashboard-v2.spec.ts`) — morning→feed→detail, reflection in feed, mobile Insights/Review render, supplement edit/remove persistence, Call/Demo gone + Train increments. They run on localhost in CI (could not run in the dev sandbox: `DATABASE_URL`/`TEST_USER_PASSWORD`/`IRON_SESSION_PASSWORD` not present — global-setup fails loudly per repo rule, validated by CI).

## Architectural review (areas of weakness)

- **`useHealthData` double-fetch** — the page now calls `useHealthData()` in addition to `MorningLogDrawer`, so health endpoints are fetched twice on load. Acceptable for now (the hook isn't context-based); a follow-up could lift it to context or fold into `useMissionStore`. *Tech-debt, not blocking.*
- **`financialData`/`focusData` typed `any`** by `useDashboardData`, so the detail resolver casts to the known `FinancialEntry`/`FocusSession` shapes. Field names verified against the real interfaces; a follow-up could type the hook to drop the casts.
- **Reflection completeness coupling** — feed "N of M answered" and the detail's per-question matching both assume question-text stability; latent only if question text is edited after answers exist. No action needed now.
- `MorningLogDrawer` is large (~940 lines); the new edit/remove UI was extracted into a reusable `EditableItemControls` rather than inlined, to avoid further bloat.

## Edge cases identified & tested

See "Failure / edge paths" above — each is covered by unit tests (missing-record fallback, partial metrics, rename validation/duplicates, template-vs-history semantics, cross-day sort) and/or the E2E persistence flows.

## Monitoring & logging

- Uniform structured `console.log('Health note template updated:', { operation, name })` on the success path of every template mutation.
- Detail-sheet lookups guard against missing records (render fallback, no throw).
- Existing API error logging on the health-notes route retained.

## PR Checklist (v2)

- [x] Was canonical Agents.md followed?
- [x] Was code coverage report included?
- [x] Was a behavior-first summary provided?
- [x] Were all user flows documented (happy path + failure paths)?
- [x] Was evidence included (screenshot, recording, or CLI output)?
- [x] Was an architectural review completed (areas of weakness identified)?
- [x] Were edge cases identified and tested?
- [x] Was monitoring and logging addressed?

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Unified activity feed consolidating morning logs, reflections, focus, and financial entries in a single chronological view with read-only detail sheets.
  * Editable template controls for renaming and removing supplements, habits, and environment fields.
  * Live Insights and Review tabs on mobile dashboard.

* **Updates**
  * Morning log icon changed from Moon to NotebookPen.
  * Removed Call and Demo quick actions; added Deep work option.

* **Documentation**
  * Added implementation plan and design specification for unified activity log feature.

### CODE REVIEW

File: src/components/dashboard/v2/ActivityFeed.tsx [hunk 1]
```diff
@@ -60,15 +66,37 @@ export function ActivityFeed({ entries }: { entries: ActivityEntry[] }) {
...
+function ActivityRow({
+  entry,
+  onOpenDetail,
+}: {
+  entry: ActivityEntry;
+  onOpenDetail?: (entry: ActivityEntry) => void;
+}) {
   const isPositive = entry.delta.startsWith('+');
+  const clickable = !!onOpenDetail;
   return (
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/ActivityFeed.tsx:82]: Fixed in 58c7da6. `ActivityRow` is now interactive only when `onOpenDetail` **and** `entry.source` are present (`clickable = !!onOpenDetail && !!entry.source`), so optimistic/local rows without a source render as plain non-interactive divs. Added an ActivityFeed test asserting a source-less entry is not clickable.

File: src/components/dashboard/v2/MorningLogDrawer.tsx [hunk 2]
```diff
@@ -401,6 +402,57 @@ function MorningLogBody({ onClose }: { onClose: () => void }) {
     }
   }, [newHabitName, updateTemplate]);
 
+  const handleRemoveSupplement = useCallback(async (name: string) => {
+    const result = await updateTemplate('removeSupplement', name);
+    if (result?.success) {
+      setSupplements((prev) => prev.filter((s) => s.name !== name));
+    }
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/MorningLogDrawer.tsx:425]: Fixed in 58c7da6. The hydration builders (`buildSupplements`/`buildHabits`/`buildEnv`) now append items present in the saved note but missing from the template, after the template-derived items. So removing/renaming a template item no longer drops historical data from a past note on reopen/save — template changes affect future logs only, history is preserved.

File: src/components/dashboard/v2/derive.ts [hunk 3]
```diff
@@ -104,6 +168,8 @@ export function deriveActivity(opts: {
     ...optimistic,
     ...liveFocus.map(focusToActivity),
     ...liveFinancial.map(financialToActivity),
+    ...morning.map(morningToActivity),
+    ...reflection.map(reflectionToActivity),
   ];
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/derive.ts:179]: Fixed in 58c7da6. `deriveActivity` now applies the same `isE2EDescription` guard to the new sources: morning notes are filtered on `freeformNote`, reflections on any answer text. Added a derive test covering both exclusions.

File: src/app/dashboard/page.tsx [hunk 4]
```diff
@@ -134,6 +142,58 @@ export default function MissionControlV2Page() {
...
+          title: 'Focus session',
+          category: f?.category ?? 'Focus',
+          hours: f?.hours ?? 0,
+          description: f?.description ?? '',
+          when: fmtWhen(entry.tsMs),
+        };
+      }
+      setDetail(resolved);
+      setDetailOpen(true);
+    },
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/app/dashboard/page.tsx:195]: Fixed in 9c753a1. The resolver now fails closed for money/focus: if the entry's `refKey` isn't found in `recentEntries`/`recentSessions`, `openDetail` returns early and the sheet does not open (no synthetic $0/empty values). Morning/reflection still open since the sheet has a 'No data' fallback and the record may be loading.

File: src/components/dashboard/v2/MorningLogDrawer.tsx [hunk 5]
```diff
@@ -401,6 +402,57 @@ function MorningLogBody({ onClose }: { onClose: () => void }) {
...
+  }, [updateTemplate]);
+  const handleRenameSupplement = useCallback(async (originalName: string, newName: string) => {
+    // Read the real dosage from the TEMPLATE, not local state (local dosage is
+    // 0 when the supplement isn't "taken", which would clobber the template).
+    const tmpl = templates.supplementTemplate.find((s) => s.name === originalName);
+    const newDosageMg = tmpl?.defaultDosageMg && tmpl.defaultDosageMg > 0 ? tmpl.defaultDosageMg : 1;
+    const result = await updateTemplate('editSupplement', originalName, undefined, { newName, newDosageMg });
+    if (result?.success) {
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/MorningLogDrawer.tsx:438]: Fixed in 9c753a1. `handleRenameSupplement` now resolves dosage as template → current local dosage → 1, so a briefly-stale template no longer overwrites the dosage to 1mg on rename.

File: src/components/dashboard/v2/derive.ts [hunk 6]
```diff
@@ -82,6 +85,65 @@ function financialToActivity(e: FinancialEntryLike): ActivityEntry {
...
+  const tsMs = lastAt || parseTimestampMs(`${entry.date}T21:00:00`);
+  return {
+    id: `reflection-${entry.date}`,
+    t: hhmm(lastAt ? new Date(lastAt).toISOString() : `${entry.date}T21:00:00`),
+    tsMs,
+    kind: 'reflection',
+    delta: '',
+    label: 'Reflection',
+    meta: `${answered} of ${total} answered`,
+    source: 'reflection',
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/derive.ts:147]: Fixed in 9c753a1. `reflectionToActivity` now counts current questions that have a non-empty matching answer, so `answered` can never exceed `total`.

File: src/components/dashboard/v2/ActivityFeed.tsx [hunk 7]
```diff
@@ -60,15 +66,37 @@ export function ActivityFeed({ entries }: { entries: ActivityEntry[] }) {
...
+            'data-testid': `activity-row-${entry.id}`,
+            onClick: () => onOpenDetail(entry),
+            onKeyDown: (e: React.KeyboardEvent) => {
+              if (e.key === 'Enter' || e.key === ' ') {
+                e.preventDefault();
+                onOpenDetail(entry);
+              }
+            },
+          }
+        : {})}
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/ActivityFeed.tsx]: Fixed in 58c7da6. Rows are interactive only when `onOpenDetail` AND `entry.source` are present, so unresolved/optimistic rows render as plain non-interactive divs. Added an ActivityFeed test for the source-less case.

File: src/components/dashboard/v2/types.ts [hunk 8]
```diff
@@ -33,8 +35,15 @@ export type ActivityEntry = {
...
+  delta: string;    // '+1h', '+ Generated', 'sync', or '' for summary cards
   label: string;
   meta: string;
+  // Discriminates the entry's origin so the detail sheet knows how to
+  // resolve the full record. Optional for back-compat with existing
+  // fixtures and optimistic rows.
+  source?: ActivitySource;
+  // Lookup key for the detail sheet: YYYY-MM-DD for morning/reflection,
+  // the underlying entry id for money/focus.
+  refKey?: string;
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/types.ts:48]: Mitigated in 58c7da6. `source`/`refKey` stay optional for back-compat with optimistic rows and hand-built fixtures, but `ActivityFeed` now gates interactivity on `entry.source` (and `openDetail` bails for source-less), so a source-less row is never presented as clickable.

File: src/components/dashboard/v2/MorningLogDrawer.tsx [hunk 9]
```diff
@@ -401,6 +402,57 @@ function MorningLogBody({ onClose }: { onClose: () => void }) {
...
+      setEnv((prev) => {
+        const next = { ...prev.customFields };
+        const v = next[originalName];
+        delete next[originalName];
+        next[newName] = v ?? false;
+        return { ...prev, customFields: next };
+      });
+    }
+  }, [updateTemplate]);
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply src/components/dashboard/v2/MorningLogDrawer.tsx:475]: Mitigated in 58c7da6. The hydration builders (`buildSupplements`/`buildHabits`/`buildEnv`) now merge in items recorded on the saved note but missing from the template, so reopening/saving a past note no longer drops history. Template edits affect future blank logs; recorded history is preserved on hydrate. A fuller separation of template-management from the active-note form is noted as a follow-up.

File: tests/e2e/dashboard-v2.spec.ts [hunk 10]
```diff
@@ -498,6 +498,61 @@ test.describe('Mission Control v2 — mobile viewport', () => {
...
+      res.request().method() === 'POST' &&
+      res.status() === 200,
+    );
+    await page.getByTestId('mobile-quick-train').click();
+    await trainPost;
+
+    await expect.poll(async () => {
+      const res = await (await request.get('/api/weekly-tracker')).json();
+      return res.todaysEntry?.trained ?? null;
+    }, { timeout: 10_000 }).toBe(true);
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply tests/e2e/dashboard-v2.spec.ts]: Fixed in 9c753a1. The Train test now captures the `deleteDay` response and asserts `del.ok()` before exercising + Train, so a failed cleanup can't mask a regression via stale state.

File: tests/e2e/morning-log.spec.ts [hunk 11]
```diff
@@ -146,6 +146,134 @@ test.describe('Morning Log drawer', () => {
...
+    await expect.poll(async () => {
+      return page.evaluate(async () => {
+        const res = await fetch('/api/three-to-thrive');
+        if (!res.ok) return false;
+        const body = await res.json();
+        return (body.todaysEntry?.answers ?? []).some(
+          (a: { answer: string }) => a.answer.trim().length > 0,
+        );
+      });
+    }, { timeout: 8_000 }).toBe(true);
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply tests/e2e/morning-log.spec.ts:211]: Fixed in 9c753a1. The reflection-feed poll now matches the exact unique `phrase` the test typed (`a.answer.includes(phrase)`) instead of any non-empty answer.

File: docs/superpowers/plans/2026-06-06-unified-activity-log.md [hunk 12]
```diff
@@ -0,0 +1,1378 @@
...
+
+Change the import on line 5 from:
+```tsx
+import { Moon, Plus, X } from 'lucide-react';
+```
+to:
+```tsx
+import { NotebookPen, Plus, X } from 'lucide-react';
+```
+Then replace every `<Moon ` with `<NotebookPen ` in this file (keep the same size/props).
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply docs/superpowers/plans/2026-06-06-unified-activity-log.md]: Fixed in 9c753a1 (MD038): removed the trailing spaces inside the code spans on that line.

File: docs/superpowers/specs/2026-06-06-unified-activity-log-design.md [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+
+### Data flow after
+
+`deriveActivity()` additionally folds in **morning logs** and **reflections**:
+
+```
+deriveActivity({ focus, financial, morning, reflection, optimistic, limit })
+  → [...optimistic, focus→, financial→, morning→, reflection→]
+  → dedup by id → sort by tsMs desc → filter E2E rows → slice(limit)
+```
```
[the remark this answers is not in the document]
Comment (nadvolod) [reply docs/superpowers/specs/2026-06-06-unified-activity-log-design.md]: Fixed in 9c753a1 (MD040): added a `text` language tag to the fenced block.
